src/views/map.c

```diff
@@ -867,13 +867,13 @@ static void pop_undo(dt_view_t *self, dt_undo_type_t type, dt_undo_data_t *data)
 
 static void _push_position(dt_view_t *self, int imgid, float longitude, float latitude)
 {
-  dt_undo_geotag_t *geotag = g_malloc(sizeof(dt_undo_geotag_t));
+  dt_undo_geotag_t *geotag = malloc(sizeof(dt_undo_geotag_t));
 
   geotag->imgid = imgid;
   geotag->longitude = longitude;
   geotag->latitude = latitude;
 
-  dt_undo_record(darktable.undo, self, DT_UNDO_GEOTAG, (dt_undo_data_t *)geotag, &pop_undo);
+  dt_undo_record(darktable.undo, self, DT_UNDO_GEOTAG, (dt_undo_data_t *)geotag, &pop_undo, NULL);
```

Comment:
I don't think so, this is freed in _free_undo_data. All memory should be freed when leaving the map view.